pkg/resource/resource.go

```diff
@@ -0,0 +1,281 @@
+package resource
+
+import (
+	"fmt"
+	"github.com/appuio/seiso/cfg"
+	"github.com/appuio/seiso/pkg/kubernetes"
+	"github.com/appuio/seiso/pkg/openshift"
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sort"
+	"time"
+)
+
+type Resources interface {
+	Delete(resourceSelectorFunc cfg.ResourceNamespaceSelector) error
+	Print(batch bool)
+	GetNamespace() string
+	FilterByTime(olderThan time.Time)
+	FilterUsed() error
+	FilterByMaxCount(keep int)
+}
+
+type SecretResources interface {
+	Resources
+	LoadSecrets(listOptions metav1.ListOptions) error
+	ListSecrets() ([]string, []string, error)
+}
+
+type ConfigMapResources interface {
+	Resources
+	LoadConfigMaps(listOptions metav1.ListOptions) error
+	ListConfigMaps() ([]string, []string, error)
+}
+
+type Secrets struct {
+	*GenericResources
+}
+
+type ConfigMaps struct {
+	*GenericResources
+}
+
+type GenericResources struct {
+	KubernetesResource *[]cfg.KubernetesResource
+	Client kubernetes.CoreV1ClientInt
+	Namespace string
+}
+
+type ImagesTagsInterface interface {
+	DeleteImageTags()
+	PrintImageTags()
+}
+
+
+var (
+	predefinedResources = []schema.GroupVersionResource{
+		{Version: "v1", Resource: "pods"},
+		{Group: "apps", Version: "v1", Resource: "statefulsets"},
+		{Group: "apps", Version: "v1", Resource: "deployments"},
+		{Group: "apps.openshift.io", Version: "v1", Resource: "deploymentconfigs"},
+		{Group: "batch", Version: "v1beta1", Resource: "cronjobs"},
+		{Group: "extensions", Version: "v1beta1", Resource: "daemonsets"},
+		{Group: "extensions", Version: "v1beta1", Resource: "deployments"},
+		{Group: "extensions", Version: "v1beta1", Resource: "replicasets"},
+	}
+	helper = kubernetes.New()
```

Comment:
This client created here goes against mocking and unit testing. So what's happening here is that we create another type of client to filter out the used resources. The client that I abstracted is of type _Core_ meanwhile here is _Dynamic_, it wouldn't be so nice to add 2 clients in the [Resource](https://github.com/appuio/seiso/pull/36/files#diff-c8d851cf5fdefa80aca6ae5c50b57882R45) structure. Maybe you have some suggestions. We could create a new client abstraction on top of them.